### PR TITLE
BUGFIX: BB-1326 Cross matching uri-identifier for AMs

### PR DESCRIPTION
**Pull request by petrjanu**
1. Fixes bug BB-1326 by cross matching uri and identifiers using execution response headers. Now we are able to match AM by `identifier` even when measure was defined using `uri` in afm and vice versa. Mocks and tests for predicate factory are rewritten to cover all known scenarios.
2. Fixes typo `intentifier` => `identifier`
3. Moves couple functions from drilling and chart options builder modules to helper utilities since those functions are shared across multiple modules.
4. Fixes unreported header predicates bug that was causing inability to drill on derived measure produced from measure with enabled "show in %" option (adhoc measure).

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2478/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1878/ : FAILURE in 2m 06s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2479/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1879/ : FAILURE in 6m 46s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1000/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2495/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/586/ : UNSTABLE in 15m 13s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1003/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2499/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/588/ : SUCCESS in 21m 17s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2500/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1899/ : SUCCESS in 2m 00s

**Comment by petrjanu** [comment]
extended test - storybook

File: src/components/visualizations/chart/chartOptionsBuilder.ts [hunk 1]
```diff
@@ -34,31 +43,23 @@ import {
     parseValue,
     stringifyChartTypes
 } from '../utils/common';
-import { getChartProperties } from './highcharts/helpers';
-import { unwrap } from '../../../helpers/utils';
+import { getComboChartOptions } from './chartOptions/comboChartOptions';
```
Comment (tomassatinsky) [inline src/components/visualizations/chart/chartOptionsBuilder.ts]: seems suspicious that in this general file is imported something ComboChart-specific

Comment (tomassatinsky) [reply src/components/visualizations/chart/chartOptionsBuilder.ts]: I see - it was always there... but if there's a chance to correct it, that would be nice

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2502/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/590/ : SUCCESS in 21m 43s

File: src/factory/tests/HeaderPredicateFactory.mock.ts [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+            localIdentifier: 'measureAdhocIdentifier.localIdentifier',
+            format: '#,##0.00'
+        },
+
+        {
+            alias: 'Compare PP measure from uri-based measure',
+            localIdentifier: 'measureUri.PP.localIdentifier',
+            definition: {
+                previousPeriodMeasure: {
+                    measureIdentifier: 'measureUri.localIdentifier',
```
Comment (tomassatinsky) [inline src/factory/tests/HeaderPredicateFactory.mock.ts]: As we discussed in person `uriMeasure` would be easier to understand

Comment (tomassatinsky) [reply src/factory/tests/HeaderPredicateFactory.mock.ts]: I would maybe suggest not using the dot as it seems it has some meaning which it hasn't. 
`uriMeasureLocalIdentifier` seems better to me.

Comment (petrjanu) [reply src/factory/tests/HeaderPredicateFactory.mock.ts]: I have renamed all mocks. Hope it is more expressive and correct.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2506/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1901/ : SUCCESS in 2m 06s

**Comment by petrjanu** [comment]
> @petrjanu Is there something that can be tested by Storybook tests? For instance, some different styling based on whether an item is drillable or not? Or are the tests there and passing?

There are stories using PredicateFactory, passing.

**Comment by petrjanu** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2507/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1902/ : SUCCESS in 2m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2520/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1916/ : SUCCESS in 12m 45s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2531/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1924/ : SUCCESS in 7m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2532/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1925/ : SUCCESS in 2m 24s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2533/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1926/ : FAILURE in 5m 22s

**Comment by tomassatinsky** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2534/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1927/ : SUCCESS in 9m 37s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/543/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1928/ : SUCCESS in 2m 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1025/ : SUCCESS in 17m 45s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/504/ : SUCCESS in 23m 10s

**Pull request merged**